### ISSUE TITLE
Ignored httpx error: AttributeError: 'Client' object has no attribute '_transport'

### ISSUE THREAD

**Issue by metemaddar** [goat-community/goat#1338]
Whenever the debugging for fastAPI starts we get the following error:

```
Exception ignored in: <function Client.__del__ at 0x7fb3161a14c0>
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/httpx/_client.py", line 1134, in __del__
    self.close()
  File "/usr/local/lib/python3.9/site-packages/httpx/_client.py", line 1106, in close
    self._transport.close()
AttributeError: 'Client' object has no attribute '_transport'
```
This error seems to slow down the start/restart debugging API. As the error says I think some code is passing a Client parameter to the httpx and https tries to call ._transport.close() but the client doesn't have the _transport attribute. I think maybe it caused by other package updates.

- The httpx version at the moment is: 0.17.1
- And the updated version is: 0.23.0 , https://pypi.org/project/httpx/

I think maybe we can have this error fixed by updating the httpx package.

**Comment by metemaddar** [comment]
It seems the only package which needs httpx as dependency is cryptography which [specified needs at least version 0.21 of httpx.](https://github.com/goat-community/goat/blob/19d7a74cb98c93654d233cc1c249fdc075ce9edf/app/api/poetry.lock#L412) However the [httpx is set to install 0.17.1](https://github.com/goat-community/goat/blob/19d7a74cb98c93654d233cc1c249fdc075ce9edf/app/api/poetry.lock#L661). Maybe we need to rebuild the poetry.loc. We also use httpx inside the codes, But I think we have no conflict withe the version.

Also we have a line that specified to test [httpx version should be less than 0.19.0](https://github.com/goat-community/goat/blob/19d7a74cb98c93654d233cc1c249fdc075ce9edf/app/api/poetry.lock#L484). I don't know what it is used for.

**Comment by majkshkurti** [comment]
Yes, version 0.17.1 is the maximum because there is a dependency that we haven't upgraded to the latest version (uvicorn) which relies on some other packages that don't work with greater versions of httpx. So it will throw an error if you try to install it. 
However, I just tried to upgrade the uvicorn package to the latest version and it worked. I managed to install the latest version of httpx after and the error is gone.

**Comment by metemaddar** [comment]
Thank you @majkshkurti for the commands. How beautiful the poetry is.

**Comment by EPajares** [comment]
So we can close this issue right?

**Comment by metemaddar** [comment]
@majkshkurti Before closing would you mind commit and let me merge in order to sync branches.

**Comment by majkshkurti** [comment]
@metemaddar Yes, I will push this today.

**Comment by jaswanthm1855** [comment]
I got the same error and 
I installed uvicorn==0.20.0 and httpx==0.23.3

Will the error comes in these versions

**Comment by metemaddar** [comment]
> I got the same error and I installed uvicorn==0.20.0 and httpx==0.23.3
> 
> Will the error comes in these versions

I don't think so. The problem should get fixed at httpx 0.23.3. I didn't have any problem with that.

**Comment by jaswanthm1855** [comment]
> > I got the same error and I installed uvicorn==0.20.0 and httpx==0.23.3
> > Will the error comes in these versions
> 
> I don't think so. The problem should get fixed at httpx 0.23.3. I didn't have any problem with that.

I got it when I am using python 3.9.15, but after upgrading to 3.9.16 it is resolved. I will check if the error is still coming. 
Thank you for your quick reply on this.

**Comment by metemaddar** [comment]
> I got it when I am using python 3.9.15, but after upgrading to 3.9.16 it is resolved. I will check if the error is still coming.
> Thank you for your quick reply on this.

Thank you very much for the information.
